### PR TITLE
Interpolation

### PR DESCRIPTION
**Pull request by jmarin**
Fixes issue with calculating offset

### CODE REVIEW

File: census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
 
-import geometry._
 import feature._
-import io.geojson.FeatureJsonProtocol._
+import geometry._
 import grasshopper.census.model.AddressRange
 
 object AddressInterpolator {
 
   def calculateAddressRange(f: Feature, a: Int): AddressRange = {
```
Comment (hkeeler) [inline census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala]: Can we use more descriptive parameter names, especially for public functions.  We're using `feature` in `interpolate()`, and I think `a` is the `addressNumber`?

File: census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala [hunk 2]
```diff
@@ -38,33 +24,15 @@ object AddressInterpolator {
...
+    val pre = prefix(feature, a)
+    val sign = if (pre == "R") -1 else 1
     val line = feature.geometry.asInstanceOf[Line]
     val l = line.length
     val d = calculateDistance(range)
     val x = a - range.start
     val dist = x * l / d
-    //TODO: Review how offset is being calculated
-    val geometry = line.pointAtDistWithOffset((dist * -1), sign * 0.0001)
+    val geometry = line.pointAtDistWithOffset(dist, sign * 0.0001)
```
Comment (hkeeler) [inline census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala:39]: What is the `0.0001` value?  I believe its the offset distance from the interpolated point, but it be more clear with a well-named variable or a comment.

File: census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala [hunk 3]
```diff
@@ -91,12 +59,48 @@ object AddressInterpolator {
...
+    if (addressIsEven && rightRangeIsEven)
+      "R"
+    else if (addressIsEven && leftRangeIsEven)
+      "L"
+    else if (!addressIsEven && !rightRangeIsEven)
+      "R"
+    else if (!addressIsEven && !leftRangeIsEven)
+      "L"
+    else
+      "R"
```
Comment (hkeeler) [inline census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala:82]: Is there a legit scenario where it _should_ fall through and default to "R"?  If yes, seem comment-worthy.  If no, seems like this should either raise an Exception or log a `WARN`-level message before returning "R".  Either option should have the full details of the `Feature` so we can track down _why_ we can't determine which side of the street a given address is on.

File: census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala [hunk 4]
```diff
@@ -91,12 +59,48 @@ object AddressInterpolator {
...
+  }
+
+  private def rangeIsEven(f: Feature, isLeft: Boolean): Boolean = {
+    val prefix = if (isLeft) "L" else "R"
+    val fromEven = f.values.getOrElse(s"${prefix}FROMHN", 0)
+    val fromRange = fromEven match {
+      case "" => 0
```
Comment (hkeeler) [inline census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala:90]: Is there a legit reason for the `FROMHN`/`TOHN` properties to be present, but set to empty string?  If no, seems like another spot to log the details of the `Feature` so we can track down where/why we have bad/missing data.

Comment (jmarin) [inline census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala:90]: _Is there a legit reason for the FROMHN/TOHN properties to be present, but set to empty string?_

That case if very frequent in the TIGER dataset

**Comment by hkeeler** [comment]
@jmarin, there merge conflicts that have to be resolved before I can merge.

**Comment by jmarin** [comment]
**Do not merge, there seems to be some tests failing**

**Comment by jmarin** [comment]
@hkeeler This seems to pass now, had to add braces to an if structure and modify an integration test (geographic coordinates precision)

**Comment by hkeeler** [comment]
Looks good.  Merging...

**Pull request merged**